src/main/java/bio/terra/stairway/Flight.java

```diff
@@ -263,8 +270,23 @@ private StepResult stepWithRetry() throws InterruptedException, StairwayExecutio
       try {
         // Do or undo based on direction we are headed
         hookWrapper().startStep(flightContext);
+
         if (context().isDoing()) {
           result = currentStep.step.doStep(context());
+
+          // If we are in debug mode and a failure is set for this step AND we have not already
+          // failed here, then insert a failure. We do this right after the step completes but
```

Comment:
And the reason you only want to insert once is if there are retries?
I wonder if we should just allow that case: exhaustion of retries causing the transition to undo.
Or maybe extend the debug info to allow specifying the number of times we should insert the fault?

Reply:
SG. As discussed offline, will augment if needed.